analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -706,25 +706,40 @@ def main(args):
         json.dump(actions, f,
                   cls=log_parser.CompileCommandEncoder)
 
-    metadata = {'action_num': len(actions),
-                'command': sys.argv,
-                'versions': {
-                    'codechecker': "{0} ({1})".format(
-                        context.package_git_tag,
-                        context.package_git_hash)},
-                'working_directory': os.getcwd(),
-                'output_path': args.output_path,
-                'result_source_files': {}}
+    metadata = {
+        'version': 2,
+        'tools': [{
+            'name': 'codechecker',
+            'action_num': len(actions),
+            'command': sys.argv,
+            'version': "{0} ({1})".format(context.package_git_tag,
+                                          context.package_git_hash),
+            'working_directory': os.getcwd(),
+            'output_path': args.output_path,
+            'result_source_files': {},
+            'analyzers': {}
+        }]}
+    metadata_tool = metadata['tools'][0]
 
     if 'name' in args:
-        metadata['name'] = args.name
+        metadata_tool['run_name'] = args.name
 
     # Update metadata dictionary with old values.
     metadata_file = os.path.join(args.output_path, 'metadata.json')
     if os.path.exists(metadata_file):
         metadata_prev = load_json_or_empty(metadata_file)
-        metadata['result_source_files'] = \
-            metadata_prev['result_source_files']
+        if 'result_source_files' in metadata_prev:
+            metadata_tool['result_source_files'] = \
+                metadata_prev['result_source_files']
+        else:
+            for tool in metadata_prev.get('tools', {}):
+                if tool.get('name') != 'codechecker':
+                    continue
+
+                result_src_files = tool.get('result_source_files', {})
+                metadata_tool['result_source_files'] = \
+                    dict(metadata_tool['result_source_files'].items() +
+                         result_src_files.items())
```

Comment:
Thinking forward to Python3, check this out: https://stackoverflow.com/questions/38987/how-do-i-merge-two-dictionaries-in-a-single-expression#answer-26853961
Especially the `Critiques of Other Answers` part. A merge_dicts could be implemented here on in a util package.

Reply:
`metadata_tool['result_source_files'].update(result_src_files.items())` wouldn't be enough here?